insecure/wintermute/attackOrchestrator.go

```diff
@@ -34,7 +56,259 @@ func (o *Orchestrator) WithAttackNetwork(network insecure.AttackNetwork) {
+	if !ok {
+		return fmt.Errorf("sender of the event is not a corrupted node")
+	}
+
+	corruptedIdentity, ok := o.allIds.ByNodeID(chunkDataPackRequestEvent.CorruptedId)
+	if !ok {
+		return fmt.Errorf("could not find corrupted identity for: %x", chunkDataPackRequestEvent.CorruptedId)
+	}
+	if corruptedIdentity.Role != flow.RoleVerification {
+		return fmt.Errorf("wrong sender role for chunk data pack request: %s", corruptedIdentity.Role.String())
+	}
+
+	if o.state != nil {
+		sent, err := o.replyWithAttestation(chunkDataPackRequestEvent)
+		if err != nil {
+			return fmt.Errorf("could not reply with attestation: %w", err)
+		}
+		if sent {
+			return nil
+		}
+	}
+
+	// no result corruption yet conducted, hence bouncing back the chunk data request.
+	err := o.network.Send(chunkDataPackRequestEvent)
+	if err != nil {
+		return fmt.Errorf("could not send chunk data request: %w", err)
+	}
+
+	o.logger.Info().
+		Hex("corrupted_id", logging.ID(chunkDataPackRequestEvent.CorruptedId)).
+		Str("protocol", insecure.ProtocolStr(chunkDataPackRequestEvent.Protocol)).
+		Str("channel", string(chunkDataPackRequestEvent.Channel)).
+		Uint32("targets_num", chunkDataPackRequestEvent.TargetNum).
+		Str("target_ids", fmt.Sprintf("%v", chunkDataPackRequestEvent.TargetIds)).
+		Msg("chunk data pack request event passed through")
+
+	return nil
+}
+
+// handleChunkDataPackResponseEvent wintermutes the chunk data pack reply if it belongs to a corrupted result, and is meant to
+// be sent to an honest verification node. Otherwise, it is passed through.
+func (o *Orchestrator) handleChunkDataPackResponseEvent(chunkDataPackReplyEvent *insecure.Event) error {
+	if o.state != nil {
+		cdpRep := chunkDataPackReplyEvent.FlowProtocolEvent.(*messages.ChunkDataResponse)
+
+		lg := o.logger.With().
+			Hex("chunk_id", logging.ID(cdpRep.ChunkDataPack.ChunkID)).
+			Hex("sender_id", logging.ID(chunkDataPackReplyEvent.CorruptedId)).
+			Hex("target_id", logging.ID(chunkDataPackReplyEvent.TargetIds[0])).Logger()
+
+		// chunk data pack reply goes over a unicast, hence, we only check first target id.
+		ok := o.corruptedIds.Contains(chunkDataPackReplyEvent.TargetIds[0])
+		if o.state.containsCorruptedChunkId(cdpRep.ChunkDataPack.ChunkID) {
+			if !ok {
+				// this is a chunk data pack response for a CORRUPTED chunk to an HONEST verification node
+				// we WINTERMUTE it!
+				lg.Info().Msg("wintermuted corrupted chunk data response to an honest verification node")
+				return nil
+			} else {
+				// Illegal state! chunk data pack response for a CORRUPTED chunk to a CORRUPTED verification node.
+				// Request must have never been reached to corrupted execution node, and must have been replaced with
+				// an attestation.
+				lg.Fatal().
+					Msg("orchestrator received a chunk data response for corrupted chunk to a corrupted verification node")
+			}
+		}
+	}
+
+	// no result corruption yet conducted, hence bouncing back the chunk data request.
```

Comment:
same here: I think we can reach this code even after results were corrupted:


In my opinion, the implementation does what we desire. Just that the documentation needs to to be revised:
```suggestion
	// chunk data response is for an honest result, hence the corrupted execution node can follow the protocol and send its honest response 
```

Reply:
updated 
https://github.com/onflow/flow-go/pull/2291/commits/7b8117d079e05b99b72e0ca4fe0dfe2e0b53c43c